.github/workflows/ci.yaml

```diff
@@ -0,0 +1,77 @@
+name: ci
+on: push
+jobs:
+  build27:
+    runs-on: ubuntu-latest
+    env:
+      NOSE_SHOW_SKIPPED: 1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up python 2.7 with tox environment py27-{test,cov} on ubuntu-latest
+        uses: actions/setup-python@v2
+        with:
+          python-version: 2.7
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install base dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox sphinx
+          python -m pip install codecov || true
+      - name: Test with tox
+        run: |
+          sudo ln -s /dev/null /dev/raw1394
+          tox -e py27-test
+          tox -e py27-cov
+          codecov || true
+  build3x:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 5, 6, 7, 8 ]
+        tox-env: [ test ]
+    env:
+      NOSE_SHOW_SKIPPED: 1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up python 3.${{ matrix.python-version }} with tox environment py3${{ matrix.python-version }}-${{ matrix.tox-env }} on ubuntu-latest
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.${{ matrix.python-version }}
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install base dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox sphinx
+      - name: Test with tox
+        run: |
+          sudo ln -s /dev/null /dev/raw1394
+          if [[ ${{ matrix.python-version }} == '6' ]]; then
+              tox -e py3${{ matrix.python-version }}-test
+              tox -e py3${{ matrix.python-version }}-flake8
```

Comment:
Instead of doing flake8 here as part of the build, what do you think about doing that as a separate action? I did that for another project there:
https://github.com/sampsyo/wideq/blob/64553d52e38d094a9d362a6527286376a9d30246/.github/workflows/style.yml#L6-L10

And it's _very_ cool because it puts the linting errors inline, visible right in the PR change view.

Reply:
Sure, I think that's great!

Reply:
I looked through other actions as well. They didn't seem to include flake8 plugins like tox does. Let me know if you think we should work around this or leave as is. Maybe I missed something.

Reply:
Aha, got it. That's too bad! In the longer term, maybe we should fork one of these (such as [this one](https://github.com/TrueBrain/actions-flake8)) and add support for plugins.

Reply:
I agree, that'd be fun.